js/hang/src/publish/element.ts

```diff
@@ -3,228 +3,94 @@ import { Effect, Signal } from "@kixelated/signals";
 import { Broadcast } from "./broadcast";
 import * as Source from "./source";
 
-// TODO: remove device; it's a backwards compatible alias for source.
-// TODO remove name; it's a backwards compatible alias for path.
-const OBSERVED = ["url", "name", "path", "device", "audio", "video", "controls", "source"] as const;
+// TODO remove name; replaced with path
+const OBSERVED = ["url", "name", "path", "muted", "invisible", "source"] as const;
 type Observed = (typeof OBSERVED)[number];
 
-type SourceType = "camera" | "screen" | "file";
-
-export interface HangPublishSignals {
-	url: Signal<URL | undefined>;
-	path: Signal<Moq.Path.Valid | undefined>;
-	device: Signal<SourceType | undefined>;
-	audio: Signal<boolean>;
-	video: Signal<boolean>;
-	controls: Signal<boolean>;
-	source: Signal<SourceType | undefined>;
-	file: Signal<File | undefined>;
-}
-
-export type InstanceAvailableEvent = CustomEvent<{ instance: HangPublishInstance }>;
+type SourceType = "camera" | "screen";
 
-declare global {
-  interface GlobalEventHandlersEventMap {
-	'publish-instance-available': InstanceAvailableEvent;
-  }
-}
+// Close everything when this element is garbage collected.
+const cleanup = new FinalizationRegistry<Effect>((signals) => signals.close());
```

Comment:
This is why I think we can remove `Instance`. The cleanup isn't as immediate as the old code, but we disable everything when disconnected from the DOM so it should be good enough.